examples/api/test/cupertino/sheet/cupertino_sheet.2_test.dart

```diff
@@ -0,0 +1,26 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_api_samples/cupertino/sheet/cupertino_sheet.2.dart' as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('Tap on button displays cupertino sheet', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const example.RestorableSheetExampleApp(),
+    );
+
+    final Finder dialogTitle = find.text('Current Count');
+    expect(dialogTitle, findsNothing);
+
+    await tester.tap(find.byType(CupertinoButton));
+    await tester.pumpAndSettle();
+    expect(dialogTitle, findsOneWidget);
+
+    await tester.tap(find.text('Pop Sheet'));
+    await tester.pumpAndSettle();
+    expect(dialogTitle, findsNothing);
```

Comment:
Ideally you should also test the state restoration part, like the scenario I mentioned in my comment on this example above. You can simulate restoration in a test with [restartAndRestore](https://main-api.flutter.dev/flutter/flutter_test/WidgetTester/restartAndRestore.html), see an example of where I did that: https://github.com/flutter/flutter/pull/153723/files#diff-114784af50a13ab49960c747162f9856ddc1b9648a902296e8c50d0bc24de7c1